drivers/memc/memc_mcux_flexspi.c

```diff
@@ -72,6 +73,25 @@ int memc_flexspi_update_lut(const struct device *dev, uint32_t index,
 	return 0;
 }
 
+int memc_flexspi_update_clock(const struct device *dev,
+		enum memc_flexspi_clock_t clock)
+{
+#if CONFIG_FLASH_MCUX_FLEXSPI_HYPERFLASH
```

Comment:
I don't think this symbol would be the preferred one to check?

Reply:
`CONFIG_DT_HAS_NXP_IMX_FLEXSPI_ENABLED`? Or maybe `CONFIG_HAS_MCUX_FLEXSPI` or `CONFIG_CODE_FLEXSPI`?

Reply:
Maybe @danieldegrasse can suggest a beter way?

IMO, ideally the clocks would come from device tree bindings and get abstracted without the need for special casing in SoC code. But maybe that would take us too far in this PR.

Reply:
I agree- clocks from devicetree bindings is indeed the right way to do this. I consider the SOC function a short term implementation, since device tree based clocking is probably not something we want to hold this PR up with

Another option here would be to only call the soc clock reconfiguration function if the `SOC_SERIES_IMX_RT10XX` KConfig is set, because this is the only SOC implementing this support. I would also return `ENOTSUP` if the `imxrt_flexspi_clock_set_div` is not called, since the clock is not actually reconfigured.

One other nit: I'd prefer this function to not include the `imxrt` prefix- it is possible that other families could implement FlexSPI in the future, and need this function.